applications/RANSApplication/custom_utilities/rans_variable_difference_norm_calculation_utility.cpp

```diff
@@ -0,0 +1,97 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+// System includes
+#include <cmath>
+#include <tuple>
+
+// External includes
+
+// Project includes
+#include "includes/communicator.h"
+#include "utilities/parallel_utilities.h"
+#include "utilities/reduction_utilities.h"
+
+// Application includes
+
+// Include base h
+#include "rans_variable_difference_norm_calculation_utility.h"
+
+namespace Kratos
+{
+template <typename TDataType>
+void RansVariableDifferenceNormsCalculationUtility<TDataType>::InitializeCalculation()
+{
+    KRATOS_TRY
+
+    const auto& r_communicator = mrModelPart.GetCommunicator();
+    const auto& r_nodes = r_communicator.LocalMesh().Nodes();
+    const int number_of_nodes = r_nodes.size();
+
+    KRATOS_ERROR_IF(!mrModelPart.HasNodalSolutionStepVariable(mrVariable))
+        << mrVariable.Name() << " not found in nodal solution step variables list in "
+        << mrModelPart.Name() << ".\n";
+
+    if (static_cast<int>(mData.size()) < number_of_nodes) {
+        mData.resize(number_of_nodes);
+    }
+
+    IndexPartition<int>(number_of_nodes).for_each([&](const int iNode) {
+        const auto p_node = r_nodes.begin() + iNode;
+        mData[iNode] = p_node->FastGetSolutionStepValue(mrVariable);
+    });
+
+    KRATOS_CATCH("");
+}
+
+template <typename TDataType>
+std::tuple<double, double> RansVariableDifferenceNormsCalculationUtility<TDataType>::CalculateDifferenceNorm()
```

Comment:
Furthermore, I'd move this to the `VariableUtils` so anyone that needs to calculate a norm could use it.